scripts/dumps/dump_gene_set_from_core.pl

```diff
@@ -72,21 +79,26 @@ =head1 OPTIONS
 use Getopt::Long;
 use Pod::Usage;
 
-my ($dbname, $host, $port, $gene_set_dump_file, $help);
+my ($dbname, $host, $port, $gene_set_dump_file, $id_type, $help);
 
 GetOptions(
     'core-db|core_db=s' => \$dbname,
     'host=s'            => \$host,
     'port=s'            => \$port,
     'outfile=s'         => \$gene_set_dump_file,
+    'id_type=s'         => \$id_type,
     'h|help'            => \$help
 );
 
 pod2usage(1) if $help;
-unless ($dbname and $host and $port and $gene_set_dump_file) {
+unless ($dbname and $host and $port and $gene_set_dump_file and $id_type) {
     pod2usage(1);
 }
 
+if ( $id_type !~ /gene|protein/ ) {
```

Comment:
This regex is not specific enough, and while it doesn't match `gallus`, it would match any string containing either `gene` or `protein` (e.g. `acetobacteroides_hydrogenigenes_gca_004340205`). I strongly recommend anchoring the start and end of `$id_type` with `^` and `$` (respectively), and also grouping `gene` and `protein` together so that the regex matches either of them, but always requires the anchors outside the group (i.e. always requires a complete match).
```suggestion
if ( $id_type !~ /^(?:gene|protein)$/ ) {
```
Making the [group non-capturing](https://perldoc.perl.org/perlretut#Non-capturing-groupings) is not necessary but is good practice because we're not using the capture group.

Reply:
Totally agree. I have to admit, I did some lazy C/P-ing from `dump_genome_from_core.pl` without double-checking what the code exactly does 🙊 

I like your suggestion about the POD/Error msg as well 😺